src/cpuid.cpp

```diff
@@ -46,7 +46,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <cstring> // for std::memset
 #endif
 
-#if TORRENT_HAS_ARM
+#if TORRENT_HAS_ARM && TORRENT_HAS_ARM_NEON
```

Comment:
It is used for `TORRENT_HAS_ARM_CRC32` too, I think you can add it to this condition and solve the issue. What toolchain/sysroot are you using?